scarb/src/ops/fmt.rs

```diff
@@ -21,30 +22,32 @@ pub struct FmtOptions {
 #[tracing::instrument(skip_all, level = "debug")]
 pub fn format(opts: FmtOptions, ws: &Workspace<'_>) -> Result<bool> {
     ws.config().ui().force_colors_enabled(opts.color);
-    let config = FormatterConfig::default();
-    let fmt = CairoFormatter::new(config);
+
+    let all_correct = AtomicBool::new(true);
+
     let packages = ws
         .members()
-        .filter(|pkg| opts.packages.contains(&pkg.id))
-        .collect::<Vec<_>>();
-    let Some((first_package, packages)) = packages.split_first() else {
-        return Ok(true);
-    };
-    let mut walk = fmt.walk(first_package.root().as_std_path());
+        .filter(|package| opts.packages.contains(&package.id));
```

Comment:
nitpick: sounds like you want to use within the `for` loop https://github.com/software-mansion/scarb/blob/d98fca8190ff330109b414ed70634b569732a6b5/scarb/src/core/workspace.rs#L112-L115

note: the old code looked like this because it collected everything to vector; now you can ignore it